superset/views/core.py

```diff
@@ -298,6 +301,64 @@ class DatabaseAsync(DatabaseView):
 appbuilder.add_view_no_menu(DatabaseAsync)
 
 
+class CsvToDatabaseView(SimpleFormView):
+    form = CsvToDatabaseForm
+    form_title = _('CSV to Database configuration')
+    add_columns = ['database', 'schema', 'table_name']
+
+    def form_get(self, form):
+        form.sep.data = ','
+        form.header.data = 0
+        form.squeeze.data = False
+        form.names.data = None
+        form.mangle_dupe_cols.data = True
+        form.skipinitialspace.data = False
+        form.skip_blank_lines.data = True
+        form.parse_dates.data = True
+        form.infer_datetime_format.data = True
+        form.dayfirst.data = False
+        form.decimal.data = '.'
+        form.error_bad_lines.data = False
+        form.if_exists.data = 'append'
+        all_datasources = []
+        all_datasources = db.session.query(
+            models.Database.sqlalchemy_uri, models.Database.database_name)\
+            .all()
+        form.con.choices = all_datasources
+
+
+    def form_post(self, form):
+        def upload_file(csv_file):
+            if csv_file and csv_file.filename:
+                filename = secure_filename(csv_file.filename)
+                csv_file.save(os.path.join(config['UPLOAD_FOLDER'], filename))
+                return filename
+
+        form.names.data = form.names.data.split(",") if form.names.data else 0
```

Comment:
Seems like `form.names.data` should receive consistent types, not `list` or `Int` depending... Maybe `[]` or `None`?